src/algebra/order/nonneg.lean

```diff
@@ -117,7 +114,7 @@ rfl
 
 @[simp, norm_cast]
 protected lemma coe_nsmul [add_monoid α] [preorder α] [covariant_class α α (+) (≤)]
-  (n : ℕ) (a : {x : α // 0 ≤ x}) : ((n • a : {x : α // 0 ≤ x}) : α) = n • a := rfl
+  (a : {x : α // 0 ≤ x}) (n : ℕ) : ((n • a : {x : α // 0 ≤ x}) : α) = n • a := rfl
```

Comment:
Why? The old ordering matched the way `n` and `a` occur in both the LHS and RHS of the goal.

Reply:
This is what [`function.injective.linear_ordered_semifield`](https://leanprover-community.github.io/mathlib_docs/algebra/order/field.html#function.injective.linear_ordered_semifield) expects. It traces back to [`function.injective.add_monoid`](https://leanprover-community.github.io/mathlib_docs/algebra/group/inj_surj.html#function.injective.add_monoid) because I assume nobody cared about writing the additive versions by hand just to swap an argument in an hypothesis.

Reply:
It goes further than that; one of the nsmul / npow fields is backwards wrt the notation; again, to make things easier with to_additive.

Reply:
So do we agree that this change is reasonable, or should I leave it as it was?

Reply:
Hmm, I would rather leave it as it was, in the hope that those other examples will also be fixed in the future.